tools/webassembly.py

```diff
@@ -123,15 +125,17 @@ class Module:
   """Extremely minimal wasm module reader.  Currently only used
   for parsing the dylink section."""
   def __init__(self, filename):
+    self.filename = filename
     self.size = os.path.getsize(filename)
     self.buf = open(filename, 'rb')
     magic = self.buf.read(4)
     version = self.buf.read(4)
-    assert magic == MAGIC
-    assert version == VERSION
+    if magic != MAGIC or version != VERSION:
+      raise InvalidWasmError(f'{filename} is not a valid wasm file')
 
   def __del__(self):
-    self.buf.close()
+    if hasattr(self, 'buf'):
```

Comment:
For some reason the use of hasattr feels odd.   Maybe add `buf = None` at the class level and use `if self.buf:` here?